packages/material-ui-lab/src/ClockPicker/Clock.tsx

```diff
@@ -34,68 +34,74 @@ export interface ClockProps<TDate> extends ReturnType<typeof useMeridiemMode> {
   ) => string;
 }
 
-export const styles = (theme: Theme) =>
-  createStyles({
-    root: {
-      display: 'flex',
-      justifyContent: 'center',
-      alignItems: 'center',
-      margin: theme.spacing(2),
-    },
-    clock: {
-      backgroundColor: 'rgba(0,0,0,.07)',
-      borderRadius: '50%',
-      height: 220,
-      width: 220,
-      flexShrink: 0,
-      position: 'relative',
-      pointerEvents: 'none',
-    },
-    squareMask: {
-      width: '100%',
-      height: '100%',
-      position: 'absolute',
-      pointerEvents: 'auto',
-      outline: 0,
-      // Disable scroll capabilities.
-      touchAction: 'none',
-      userSelect: 'none',
-      '&:active': {
-        cursor: 'move',
-      },
-    },
-    pin: {
-      width: 6,
-      height: 6,
-      borderRadius: '50%',
-      backgroundColor: theme.palette.primary.main,
-      position: 'absolute',
-      top: '50%',
-      left: '50%',
-      transform: 'translate(-50%, -50%)',
-    },
-    amButton: {
-      zIndex: 1,
-      left: 8,
-      position: 'absolute',
-      bottom: 8,
-    },
-    pmButton: {
-      zIndex: 1,
-      position: 'absolute',
-      bottom: 8,
-      right: 8,
+export type ClockClassKey =
+  | 'root'
+  | 'clock'
+  | 'squareMask'
+  | 'pin'
+  | 'amButton'
+  | 'pmButton'
+  | 'meridiemButtonSelected';
+
+export const styles: MuiStyles<ClockClassKey> = (theme): StyleRules<ClockClassKey> => ({
```

Comment:
This might seem like duplication but leads to better locations of type errors of classes or not implemented or not declared (compared to only using `MuiStyles`; just using `Theme => StyleRules` would not allow us to switch back to theme-less styles).